test/performance/odp_timer_perf.c

```diff
@@ -612,53 +848,68 @@ static void print_stat(test_global_t *global)
-		after.num += global->stat[i].after.num;
-		after.sum_ns += global->stat[i].after.sum_ns;
-
-		if (global->stat[i].before.max_ns > before.max_ns)
-			before.max_ns = global->stat[i].before.max_ns;
-
-		if (global->stat[i].after.max_ns > after.max_ns)
-			after.max_ns = global->stat[i].after.max_ns;
-	}
+	if (sum->num)
+		round_ave = (double)sum->rounds / sum->num;
 
-	if (num) {
-		time_ave  = ((double)time_sum / num) / ODP_TIME_SEC_IN_NS;
-		round_ave = (double)round_sum / num;
-	}
+	if (sum->before.num)
+		before_ave = (double)sum->before.sum_ns / sum->before.num;
 
-	if (before.num)
-		before_ave = (double)before.sum_ns / before.num;
+	if (sum->after.num)
+		after_ave = (double)sum->after.sum_ns / sum->after.num;
 
-	if (after.num)
-		after_ave = (double)after.sum_ns / after.num;
-
-	printf("TOTAL (%i workers)\n", num);
-	printf("  events:             %" PRIu64 "\n", event_sum);
-	printf("  ave time:           %.2f sec\n", time_ave);
-	printf("  ave rounds per sec: %.2fM\n",
-	       (round_ave / time_ave) / 1000000.0);
-	printf("  num before:         %" PRIu64 "\n", before.num);
+	printf("TOTAL (%i workers)\n", sum->num);
+	printf("  events:             %" PRIu64 "\n", sum->events);
+	printf("  ave time:           %.2f sec\n", sum->time_ave);
+	printf("  ave rounds per sec: %.2fM\n", (round_ave / sum->time_ave) / 1000000.0);
+	printf("  num before:         %" PRIu64 "\n", sum->before.num);
 	printf("  ave before:         %.1f nsec\n", before_ave);
-	printf("  max before:         %" PRIu64 " nsec\n", before.max_ns);
-	printf("  num after:          %" PRIu64 "\n", after.num);
+	printf("  max before:         %" PRIu64 " nsec\n", sum->before.max_ns);
+	printf("  num after:          %" PRIu64 "\n", sum->after.num);
 	printf("  ave after:          %.1f nsec\n", after_ave);
-	printf("  max after:          %" PRIu64 " nsec\n", after.max_ns);
+	printf("  max after:          %" PRIu64 " nsec\n", sum->after.max_ns);
+	printf("\n");
+}
+
+static void print_stat_set_cancel_mode(test_global_t *global)
+{
+	int i;
+	test_stat_sum_t *sum = &global->stat_sum;
+	double set_ave = 0.0;
+	int num = 0;
+
+	printf("\n");
+	printf("RESULTS - timer cancel + set cycles per thread:\n");
+	printf("-----------------------------------------------\n");
+	printf("        1      2      3      4      5      6      7      8      9     10");
+
+	for (i = 0; i < ODP_THREAD_COUNT_MAX; i++) {
+		if (global->stat[i].rounds) {
+			if ((num % 10) == 0)
+				printf("\n   ");
+
+			printf("%6.1f ", (double)global->stat[i].cycles / global->stat[i].sets);
+			num++;
+		}
+	}
+
+	if (sum->num)
+		set_ave = (double)sum->sets / sum->num;
+
+	printf("\n\n");
+	printf("TOTAL (%i workers)\n", sum->num);
+	printf("  rounds:              %" PRIu64 "\n", sum->rounds);
+	printf("  timeouts:            %" PRIu64 "\n", sum->events);
+	printf("  timer cancels:       %" PRIu64 "\n", sum->cancels);
+	printf("  timer sets:          %" PRIu64 "\n", sum->sets);
```

Comment:
Perhaps it could be useful to print the number of timer cancel failures. In practice it's `cancels - sets`, but a separate print would be clearer to a user.

Reply:
fixed in v6